### PR TITLE
Add FilterTypes enum, and filter by type and ability …

### LINKED ISSUE #39: Allow user to sort and filter
**Is your feature request related to a problem? Please describe.**
Currently user can only either filter or sort, not filter and sort at the same time.

**Describe the solution you'd like**
Refactor to allow user to filter and sort.

### PR DESCRIPTION
**Pull request by connielion**
## Changes
1. Create new enum for filter types in FilterTypes.js.
2.  Add conditional to check if `selectTypeOption` and `selectAbilityOption` is empty string and array as initialized in hook in `sortTypeCheck`.
3. Added 'none' label option with empty string as value in Select drop-downs.
4.  Added select option state variables into useEffect dependency array.

## Purpose
Allow multiple filter and sorting.

## Approach
Use state hooks to set user's options for filtering, and s

Closes #39

### CODE REVIEW

File: src/components/Filter.js [hunk 1]
```diff
@@ -62,20 +64,21 @@ function Filter({
 
   return (
     <div className="filterBar">
-      <div>
         <Select
           className="filterDropDown"
           onChange={handleTypeChange}
-          options={allTypes}
+          // onClick={()=>setFilterType(FilterTypes.TYPE)}
```
Comment (krferrell) [inline src/components/Filter.js:70]: Is this going to be used in the future?

File: src/components/Filter.js [hunk 2]
```diff
@@ -62,20 +64,21 @@ function Filter({
...
+          options={[{value: '', label: 'None'},...allTypes]}
           placeholder="TYPE"
           styles={dropDownStyles}
         />
-      </div>
+
       <Select
         placeholder="ABILITY"I
         className="filterDropDown"
+        // onClick={()=>setFilterType(FilterTypes.ABILITY)}
```
Comment (krferrell) [inline src/components/Filter.js:79]: Here too?

File: src/components/Filter.js (hunk 1 above)
Comment (kmausolf) [inline src/components/Filter.js:70]: Should remove these commented out sections.

File: src/components/PokeList.js [hunk 3]
```diff
@@ -192,10 +205,13 @@ const PokeList = ({
...
             setAbilityOptions={setAbilityOptions}
             setSortType={setSortTypeFlags}
+            setFilterType={setFilterType}
           />
           <div className="card-container">
-          {/* By default, display all Pokemons, else show  filtered/sorted cards */}
-            {generateSortedCards(userDidSort ? filteredPokemons : allPokemons)}
+            {/* By default, display all Pokemons, else show  filtered/sorted cards */}
+            {generateSortedCards(
+              userDidSort || userDidFilter ? filteredPokemons : allPokemons
```
Comment (krferrell) [inline src/components/PokeList.js:213]: Is it necessary to generate sorted cards of allPokemons? I notice above that you set allPokemonsFiltered to be allPokemons and then set filteredPokemons to allPokemonsFiltered. Wouldn't allPokemons = filteredPokemons at first and then when filter the data, allPokemonsFiltered gets updated? It is hard to tell without seeing all the code. Disregard this if it is totally wrong.

**Review by krferrell — APPROVED** [review]
Lots of great work here! The filtering and sorting are mind-boggling.

**Review by BlackCubes — APPROVED** [review]
Looks good. Super excited to see this in action 🎉!

File: src/components/PokeList.js (hunk 3 above)
Comment (kmausolf) [reply src/components/PokeList.js]: Good question.  Also, I'm not sure there's a need to differentiate between whether the user sorted or filtered.  Either way, it derives `allPokemonsFiltered`.

**Pull request merged**